dask/bag/core.py

```diff
@@ -244,20 +267,53 @@ def __init__(self, dsk, name, npartitions):
         self.npartitions = npartitions
         self.str = StringAccessor(self)
 
-    def map(self, func):
+    def map(self, func, **kwargs):
         """ Map a function across all elements in collection
 
         >>> import dask.bag as db
         >>> b = db.from_sequence(range(5))
         >>> list(b.map(lambda x: x * 10))  # doctest: +SKIP
         [0, 10, 20, 30, 40]
+
+        Any **kwargs are passed through to func along with the (maybe unpacked)
+        item argument(s):
+
+        >>> import dask.bag as db
+        >>> b = db.from_sequence(range(100), npartitions=10)
+        >>> def normalize(x, total=0):
+        ...     return x / total
+        >>> total = b.sum().compute()
+        >>> total
+        4950
+        >>> normalized = b.map(normalize, total=total)
+        >>> normalized.sum().compute()
+        1.0
+
+        If a dask.Bag.Item is used its computed value is passed to the map
+        function.  This means that whatever computations are implied by any
+        Item values block any of the map work.
+
+        For example the above ended up doing its two passes over the data in
+        two separate dask computations.  If the data were expensive to access
+        (rather than being an in-memory range) we would instead want to combine
+        both passes over the data into one shared computation:
+
+        >>> normalized = b.map(normalize, total=b.sum())
+        >>> normalized.sum().compute()
+        1.0
```

Comment:
This docstring took me a bit to fully grok, which means it could probably be cleaner. Perhaps:

```python
"""
...

Keyword arguments are passed through to `func`. These can be either 
`dask.bag.Item`s, or normal python objects.

Example
-------
>>> b = db.from_sequence(range(1, 101), npartitions=10)
>>> def div(num, den=1):
...     return num/den

Using a python object:

>>> den = b.max().compute()
>>> den
100
>>> b.map(div, den=den).take(5)
[0.01, 0.02, 0.03, 0.04, 0.05]

Using an `Item`:

>>> b.map(div, den=b.max()).take(5)
[0.01, 0.02, 0.03, 0.04, 0.05]

Note that while both versions give the same output, the second forms 
a single graph, and then computes everything at once, and in some 
cases may be more efficient.
"""
```